nixos/modules/services/networking/charybdis.nix

```diff
@@ -21,30 +21,30 @@ in
       enable = mkEnableOption "Charybdis IRC daemon";
 
       config = mkOption {
-        type = types.string;
+        type = types.str;
         description = ''
           Charybdis IRC daemon configuration file.
         '';
       };
 
       statedir = mkOption {
-        type = types.string;
+        type = types.str;
```

Comment:
I wanted to suggest `types.path`, but then I've recalled about a problem this may introduce - https://github.com/NixOS/nixpkgs/pull/57168.

So maybe some new type `types.strPath` or `types.statePath` should be introduced, which is `either str path` but does check if string starts with `/`, and does `toString path` if path object was provided.